src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1207 @@
+            if not coefficients:
+                for name in names:
+                    # variables present in this polynomial
+                    try:
+                        name_poly_vars = set(SR(data[name]).variables())
+                    except BaseException:
+                        raise TypeError('values of dictionary are not polynomials')
+                    # add the variables that are not already names
+                    coefficients += tuple(name_poly_vars.difference(set(names)))
+
+            coefficients = tuple(coefficients)  # coefficients are immutable
+            exchange_polys = list(data.values())
+
+            self._names = names
+
+            # construct the ambient ring from input data
+
+            # first we get generators for ambient ring
+            variables = names + coefficients
+
+            # now attempt to build polynomial ring from this data
+            if not isinstance(base_ring, (IntegerRing_class, RationalField)):
+                raise ValueError('%s is a nonsupported base ring. ZZ or QQ only.' %
+                                 (base_ring))
+
+            self._base_ring = base_ring
+            try:
+                self._polynomial_ring = PolynomialRing(self._base_ring,
+                                                       names=variables)
+            except BaseException:
+                raise TypeError('supplied variables / coefficients are not correct')
+
+            self._ambient_field = FractionField(self._polynomial_ring)
+            self._rank = len(self._names)
+
+            # we get initial cluster variables by casting initial variables as
+            # rational functions
+            self._cluster_vars = [self._ambient_field(self._names[i])
+                                  for i in range(0, self._rank)]
+
+            # take what input data we were given and try to use it to
+            # construct polynomials
+            self._exchange_polys = []
+            try:
+                for i in range(self._rank):
+                    self._exchange_polys.append(
+                        self._polynomial_ring(exchange_polys[i]))
+            except BaseException:
+                raise TypeError(("the supplied exchange polynomials are not"
+                                 " polynomials in the cluster variables over %s"
+                                 % (base_ring)))
+
+            self._coefficients = coefficients
+            self._check_seed()
+
+            # ensure mutation sequence / seed list initialises as empty lists
+            self._mutation_sequence = []
+
+            # begin with correct Laurent polynomials
+            self._laurent_polys = self._exchange_polys.copy()
+            self._compute_laurent()
+
+        else:
+            raise TypeError('nonsupported input data. See documentation.')
+
+    # a private method to do some mathematical checks on our seed
+
+    def _check_seed(self):
```

Comment:
I would consider renaming this to `_test_seed(self, **options)` as then it gets automatically called by by the `TestSuite(S).run()` .